drake/examples/Pendulum/runEnergyShaping.cpp

```diff
@@ -7,14 +5,11 @@
 using namespace std;
 
 int main(int argc, char* argv[]) {
-  Pendulum p;
   shared_ptr<lcm::LCM> lcm(new lcm::LCM);
   if(!lcm->good())
     return 1;
 
-  p.input_frame = make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumInput",p.input_frame->getCoordinateNames(),lcm);
-  p.output_frame = make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumState",p.output_frame->getCoordinateNames(),lcm);
-
+  Pendulum p(lcm);
   DrakeSystemPtr controller = make_shared<PendulumEnergyShaping>(p);
```

Comment:
ok, so all of this is a bit tricky.  As you know, shared_ptrs are a bit infectious.  I've now changed runLCM to take an lcm::LCM& (instead of a shared_ptr), but making it take a const DrakeSystem& instead of a DrakeSystemPtr is tough.  Inside the method it does things like combine lcm input/output systems with this system making new systems.  While we do know in this instance that the lifetime of those new systems is limited, all of the system combinations (like FeedbackSystem and CascadeSystem) traffic in DrakeSharedPtrs because in general this is not true.  So it would be hard to change here, and I think the cost is small.